lib/src/test/java/io/stormbird/token/tools/TokenScriptValidatorTest.java

```diff
@@ -0,0 +1,27 @@
+package io.stormbird.token.tools;
+
+import org.junit.Test;
+
+import java.io.FileInputStream;
+import java.io.InputStream;
+
+import io.stormbird.token.entity.TSMLValidationResult;
+
+public class TokenScriptValidatorTest {
+
+    @Test
+    public void verifyRSAxmldsig() throws Exception {
+        InputStream EntryToken = new FileInputStream("src/test/ts/EntryToken.tsml");
+        TSMLValidationResult result = TokenScriptValidator.validateXML(EntryToken);
+        //Should not pass because shong.wang is signed by an authority which is not recognised (cacert.org)
+        assert(!result.isValid);
+    }
+
+    @Test
+    public void verifyECDSAxmldsig() throws Exception {
+        InputStream DAIToken = new FileInputStream("src/test/ts/DAI.tsml");
+        TSMLValidationResult result = TokenScriptValidator.validateXML(DAIToken);
+        assert(result.isValid);
+        assert(result.subjectPrincipal.equals("CN=*.aw.app"));
+    }
```

Comment:
I would stress test the solution a bit more by adding some negative tests, e.g. error in the certificate chain.